### PR TITLE
abuild: init at 3.7.0

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
This pull request adds Alpine Linux abuild tool. With it one could build and package APKBUILD files in NixOS without any problems.

I'm still having some issues with patchShebangs in case of `bin/abuild` which uses `ash` as interpreter. Also the `chmod` command of `make install` doesn't seem to work, so the binaries are not executable yet :(

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136089` at 6424267d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>abuild</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/abuild/default.nix:34:3:
```
   |
34 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/tools/abuild/default.nix:28:3:
```
   |
28 |   patchPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/abuild/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  patchPhase = ''
+    substituteInPlace ./Makefile \
+      --replace '-Werror' '-Wno-error' \
+      --replace 'chmod' 'echo'
+  '';
+
+  installPhase = ''
+    DESTDIR=$out make install
+  '';
```
Comment (figsoda) [inline pkgs/development/tools/abuild/default.nix:36]: these can be replaced with `makeFlags`

File: pkgs/development/tools/abuild/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      --replace "/bin/ash" "${busybox}/bin/ash"
+    patchShebangs $out/bin/abuild
+  '';
+
+  meta = with lib; {
+    description = "Alpine Linux build tools";
+    homepage    = "https://gitlab.alpinelinux.org/alpine/abuild";
+    license     = licenses.gpl2Only;
+    maintainers = with maintainers; [ onny ];
+    platforms   = with platforms; unix;
```
Comment (figsoda) [inline pkgs/development/tools/abuild/default.nix:51]: not sure if `gpl2Only` is right here, upstream license has an openssl exception
```suggestion
    homepage = "https://gitlab.alpinelinux.org/alpine/abuild";
    license = licenses.gpl2Only;
    maintainers = with maintainers; [ onny ];
    platforms = platforms.unix;
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/abuild/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  patchPhase = ''
+    substituteInPlace ./Makefile \
+      --replace '-Werror' '-Wno-error'
+  '';
+
+  preInstall = "export DESTDIR=$out";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/abuild/default.nix:36]: Can you try the following:

```suggestion
  installFlags = [ "DESTDIR=($out)" ]]
```

File: pkgs/development/tools/abuild/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  postInstall = ''
+    wrapProgram $out/bin/abuild \
+      --prefix PATH : ${lib.makeBinPath [ busybox ]} \
+    substituteInPlace $out/bin/abuild \
```
Comment (SuperSandro2000) [inline pkgs/development/tools/abuild/default.nix:41]: If you add BusyBox to buildInputs that line can be removed.

File: pkgs/development/tools/abuild/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  patchPhase = ''
+    substituteInPlace ./Makefile \
+      --replace '-Werror' '-Wno-error' \
+      --replace 'chmod' 'echo'
+  '';
+
+  installFlags = [
+    "DESTDIR=$(out)"
+    "prefix="
+  ];
```
Comment (jtojnar) [inline pkgs/development/tools/abuild/default.nix:41]: `DESTDIR` should not be used, see https://github.com/NixOS/nixpkgs/issues/65718.

```suggestion
  makeFlags = [
    "prefix=${placeholder "out"}"
  ];

  installFlags = [
    "sysconfdir=${placeholder "out"}/etc"
  ];
```

File: pkgs/development/tools/abuild/default.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  patchPhase = ''
+    substituteInPlace ./Makefile \
+      --replace '-Werror' '-Wno-error' \
+      --replace 'chmod' 'echo'
```
Comment (jtojnar) [inline pkgs/development/tools/abuild/default.nix:35]: The issue is more likely https://gitlab.alpinelinux.org/alpine/abuild/-/blob/10ad48d1feb4a7a663674f4ad41003f03cf20149/Makefile#L107 – Nix store cannot contain SETUID files.

File: pkgs/development/tools/abuild/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  installFlags = [
+    "DESTDIR=$(out)"
+    "prefix="
+  ];
+
+  meta = with lib; {
+    description = "Alpine Linux build tools";
+    homepage = "https://gitlab.alpinelinux.org/alpine/abuild";
+    license = licenses.gpl2;
```
Comment (jtojnar) [inline pkgs/development/tools/abuild/default.nix:46]: `gpl2` is unclear, use `gpl2Plus` or `gpl2Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
Thank you so much for the helpful comments, this was very educational :)
MR looks good now for me

**Review by jtojnar — APPROVED** [review]
code lgtm

**Review by figsoda — APPROVED** [review]

**Pull request merged**